giskard/scanner/robustness/base_numerical_detector.py

```diff
@@ -0,0 +1,142 @@
+
+    def _detect_issues(
+        self,
+        model: BaseModel,
+        dataset: Dataset,
+        feature: str,
+    ) -> Sequence[Issue]:
+        num_samples = self.num_samples or min(1000, len(dataset.df))
+        output_sensitivity = self.output_sensitivity or 0.05
+        threshold = self.threshold or 0.05
+
+        # Generate perturbed dataset by adding a small percentage of change
+        perturbation = dataset.df[feature] * self.perturbation_fraction
+        perturbed_data = dataset.df.copy()
+        perturbed_data[feature] += perturbation
+
+        # Subset the dataset for faster calculations
+        perturbed_data = perturbed_data.sample(n=num_samples, random_state=42)
+        original_data = dataset.df.loc[perturbed_data.index]
+
+        # Ensure column types are passed correctly as a dictionary
+        original_pred = model.predict(
+            Dataset(df=original_data, target=dataset.target, column_types=dict(dataset.column_types))
+        )
+        perturbed_pred = model.predict(
+            Dataset(df=perturbed_data, target=dataset.target, column_types=dict(dataset.column_types))
+        )
+
+        # Check model type and calculate pass/fail rate
+        if model.is_classification:
+            passed = original_pred.raw_prediction == perturbed_pred.raw_prediction
+        elif model.is_regression:
+            rel_delta = np.abs(
+                (perturbed_pred.raw_prediction - original_pred.raw_prediction) / original_pred.raw_prediction
+            )
+            passed = rel_delta < output_sensitivity
+        else:
+            raise NotImplementedError("Only classification and regression models are supported.")
+
+        pass_rate = passed.mean()
+        fail_rate = 1 - pass_rate
+
+        logger.info("Testing `%s` perturbation\tFail rate: %.3f" % (feature, fail_rate))
+
+        issues = []
+        if fail_rate >= threshold:
+            # Severity
+            issue_level = IssueLevel.MAJOR if fail_rate >= 2 * threshold else IssueLevel.MEDIUM
+
+            # Issue description
+            desc = (
+                "When the feature `%s` is perturbed by %.2f%%, the model changes its prediction in %.2f%% of cases."
+                % (feature, self.perturbation_fraction * 100, fail_rate * 100)
+            )
+
+            failed_size = (~passed).sum()
+            slice_size = len(passed)
+
+            issue = Issue(
+                model,
+                dataset,
+                group=self._issue_group,
+                level=issue_level,
+                description=desc,
+                features=[feature],
+                meta={
+                    "feature": feature,
+                    "perturbation_fraction": self.perturbation_fraction,
+                    "fail_rate": fail_rate,
+                    "failed_size": failed_size,
+                    "slice_size": slice_size,
+                    "threshold": threshold,
+                    "output_sensitivity": output_sensitivity,
```

Comment:
add params `metric`, `metric_value` (these are used in standardized export, e.g. AVID report)

```suggestion
                    "output_sensitivity": output_sensitivity,
                    "metric": "Fail rate",
                    "metric_value": fail_rate,                    
```